rewrite-maven/src/main/java/org/openrewrite/maven/tree/MavenResolutionResult.java

```diff
@@ -203,10 +204,7 @@ public boolean parentPomIsProjectPom() {
         if (getParent() == null) {
             return false;
         }
-        ResolvedGroupArtifactVersion parentGav = getParent().getPom().getGav();
-        return getProjectPoms().values().stream()
-                .map(Pom::getGav)
-                .anyMatch(gav -> gav.equals(parentGav));
+        return getParent().getPom().getRequested().getSourcePath() != null;
```

Comment:
I first considered adding the `ExecutionContext` as a parameter but that would break the public API of `getProjectPoms`. Is `getSourcePath() != null` a good enough invariant ? if not, how could I access the `projectPoms` from here ?

Reply:
Additional tests show its not a problem.

Reply:
the way to access the parent is aligned with other implementations, for the nullness see my comments above but it seems to be no problem.

Reply:
thanks for checking and I'm glad it works out